lib/Service/OAuthService.php

```diff
@@ -0,0 +1,121 @@
+     */
+    public function authorizeUrl($params = null, $opts = null)
+    {
+        $params = $params ?: [];
+
+        $opts = \Stripe\Util\RequestOptions::parse($opts);
+        $base = $this->_getBase($opts);
+
+        $params['client_id'] = $this->_getClientId($params);
+        if (!\array_key_exists('response_type', $params)) {
+            $params['response_type'] = 'code';
+        }
+        $query = \Stripe\Util\Util::encodeParameters($params);
+
+        return $base . '/oauth/authorize?' . $query;
+    }
+
+    /**
+     * Use an authoriztion code to connect an account to your platform and
+     * fetch the user's credentials.
+     *
+     * @param null|array $params
+     * @param null|array $opts
+     *
+     * @throws \Stripe\Exception\OAuth\OAuthErrorException if the request fails
+     *
+     * @return StripeObject object containing the response from the API
+     */
+    public function token($params = null, $opts = null)
+    {
+        $params = $params ?: [];
+        $params['client_secret'] = $this->_getClientSecret($params);
+
+        return $this->requestConnect('post', '/oauth/token', $params, $opts);
+    }
+
+    public function deauthorize($params = null, $opts = null)
+    {
+        $params = $params ?: [];
+        $params['client_id'] = $this->_getClientId($params);
+
+        return $this->requestConnect('post', '/oauth/deauthorize', $params, $opts);
+    }
+
+    private function _getClientId($params = null)
+    {
+        $clientId = ($params && \array_key_exists('client_id', $params)) ? $params['client_id'] : null;
+
+        if (null === $clientId) {
+            $clientId = $this->client->getClientId();
+        }
+        if (null === $clientId) {
+            $msg = 'No client_id provided. (HINT: set your client_id using '
+              . '`new \Stripe\StripeClient([clientId => <CLIENT-ID>
+                ])`)".  You can find your client_ids '
+              . 'in your Stripe dashboard at '
+              . 'https://dashboard.stripe.com/account/applications/settings, '
+              . 'after registering your account as a platform. See '
+              . 'https://stripe.com/docs/connect/standard-accounts for details, '
+              . 'or email support@stripe.com if you have any questions.';
+
+            throw new \Stripe\Exception\AuthenticationException($msg);
+        }
+
+        return $clientId;
+    }
+
+    private function _getClientSecret($params = null)
+    {
+        if (\array_key_exists('client_secret', $params)) {
+            return $params['client_secret'];
+        }
+
+        return $this->client->getApiKey();
+    }
+
+    private function _getBase($opts)
+    {
+        // Throw an exception for the convenience of anybody migrating to
+        // \Stripe\Service\OAuthService from \Stripe\OAuth, where `connect_base`
+        // was the name of the parameter that behaves as `api_base` does here.
+        if (isset($opts->connect_base)) {
```

Comment:
I'm not sure this will work. `\Stripe\RequestOptions` does not have a `connect_base` property, even if there is a `connect_base` key in the array passed to `RequestOptions::parse()`, so I think this condition will always be false. Mind adding a test for it?

Reply:
Oops! Good catch. Fixed now.